softvis3d-frontend/src/components/scene/visualization/SoftVis3dScene.ts

```diff
@@ -150,6 +149,15 @@ export default class SoftVis3dScene {
         return result;
     }
 
+    /**
+     * Why recalculate?
+     * I want the position to be reset based on the given shapes. In the use case of the "direct link" feature,
+     * reset to the "first" position would reset to the starting point from the link.
```

Comment:
Also the position will be correct after the shapes were changed and the objects were updated in the scene directly. 👍